README.md

```diff
@@ -1,130 +1,182 @@
+4. Once they've selected their vehicles & permissions Smartcar's services will redirect to the Smartcar hosted redirect URI. This page will load `redirect.js` which will post a message to the original client page that opened the redirect. The `Smartcar` class instantiated on the client page handles this redirect, calling the required `onComplete` method provided to the constructor. The `onComplete` method you provided must handle an error if the user refused authorization or sending the code to your backend to exchange for an access token if the user authorized your application.
+5. Your application's backend server will need to accept the authorization code and exchange it for an access token.
+
+<!-- depending on timing this probably won't make it to initial launch of redirect scheme -->
+For a more detailed explanation of our Smartcar hosted redirect scheme see our [blog post](TODO: LINK HERE?) on the topic.
+
+#### Self Hosted
+
+Here's an example flow should you host the redirect yourself (many of the steps remain similar):
 
 1. User clicks "Connect your car" button (or similar) on your application's website.
-2. The user is redirected to a new page, either as a popup (if using the `openDialog` and/or `addClickHandler` methods), or in the same page (if the client redirects to the link generated by `generateLink` directly, without using `openDialog` and/or `addClickHandler`). This page requires the user authenticate with their vehicle credentials.
-3. The user will be asked to authorize your application to connect to their vehicle.
-4. Smartcar's services will redirect back to your application's redirectUri. The redirect will include an authorization code if the user approved the authorization, an error otherwise. An optional callback function (passed in as `onComplete`) will be invoked if `callback.js` is imported in the page served at your application's redirectUri.
+2. The user is redirected to an OAuth authentication page, using the `openDialog`, `addClickHandler` or `generateLink` methods. This page requires the user to authenticate with their vehicle credentials.
+3. After entering credentials the user will be presented with a list of their vehicles to authorize and a selection of permissions to allow your app.
+4. Smartcar's services will redirect back to your application's redirect URI. The redirect URI will include an authorization code query parameter if the user approved the authorization, an error query parameter otherwise. If you import the `redirect.js` script at the redirect URI it will trigger execution of your optionally provided `onComplete` method (passed to the `Smartcar` constructor) and close out the redirect page. **Note:** in the Smartcar hosted scheme `redirect.js` handles extracting the query parameters from the redirect URI. In a manual hosting scheme you can choose to do this extraction client (making use of `redirect.js` if you'd like) or server side.
```

Comment:
What does - 'If you import the `redirect.js` script at the redirect URI' mean? Typo?